### ISSUE TITLE
ALTER COLUMN TYPE fails with TABLESPACE error if column has index on it

### ISSUE THREAD

**Issue by ddorian** [yugabyte/yugabyte-db#6113]
Steps::::::

```
yugabyte=# create table bum(id bigint primary key, b varchar(50));
yugabyte=# alter table bum alter column b type varchar(100);
yugabyte=# create index on bum(b);
yugabyte=# alter table bum alter column b type varchar(150);
ERROR:  TABLESPACE not supported yet
HINT:  See https://github.com/YugaByte/yugabyte-db/issues/1129. Click '+' on the description to raise its priority
```

**Comment by danports** [comment]
Just ran into this. Note that the error is raised even if the old and new data types are the same (i.e. the ALTER COLUMN is a no-op).

**Comment by m-iancu** [comment]
Fixed by acd9b0f.
